ome_zarr/reader.py

```diff
@@ -484,8 +486,9 @@ def get_pyramid_lazy(self, node: Node) -> None:
 
         LOGGER.debug("img_pyramid_shapes", well_spec.img_pyramid_shapes)
 
-        size_y = well_spec.img_shape[3]
-        size_x = well_spec.img_shape[4]
+        self.axes = well_spec.img_metadata["axes"]
+        size_y = well_spec.img_shape[len(self.axes) - 2]
```

Comment:
How close do you think we are getting to wanting accessors for these?

Reply:
Definitely feels like something the `Axes` object proposed in #124 could handle i.e. `self.axes.get_y_index()` or similar?

Reply:
Thinking more about it, in https://github.com/ome/ome-zarr-py/pull/148/files#diff-b50d9715cc6e4017cfc055fd0ed73ecb5d9158e17f4d58ca5b3ba08b89c46657R581, I used `c_index = self.axes.index("c")` to avoid relying on magic numbers.

With the current spec, this is also a strategy that could be used here i.e. `self.axes.index("y")`. However this will also be affected by #124 since the axes representation will become a list of dictionaries rather than a list of strings. @will-moore  It looks like we need to decide on a resolution order as this PRs are becoming effectively coupled.